packages/smooth_app/lib/cache/cache_manager.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/foundation.dart';
+
+/// Abstract class defining a cache containing items of type [T] and accessible
+/// via a unique key of type [K]
+abstract class CacheManager<K, T> extends ChangeNotifier {
```

Comment:
It's not a `CacheManager`, it's rather a `CacheMap`. I would also put the `ChangeNotifier` directly in the children, not there.
And it looks like you never call `notifyListeners`, therefore I'm really not convinced about putting it there.